extensions/amp-date-display/0.1/amp-date-display.js

```diff
@@ -0,0 +1,269 @@
+          !isNaN(epoch),
+          'amp-date-display',
+          `Invalid date: '${this.datetime_}'`
+      );
+    } else if (this.timestampMiliseconds_) {
+      epoch = this.timestampMiliseconds_;
+    } else if (this.timestampSeconds_) {
+      epoch = this.timestampSeconds_ * 1000;
+    }
+
+    user().assert(
+        epoch !== undefined,
+        'amp-date-display',
+        'One of datetime, timestamp-ms, or timestamp-seconds is required'
+    );
+
+    return epoch;
+  }
+
+  /**
+   * @param {!Date} date
+   * @param {string} locale
+   * @private
+   */
+  getVariablesInLocal_(date, locale) {
+    return {
+      year: date.getFullYear(),
+      month: date.getMonth() + 1,
+      monthName: date.toLocaleDateString(locale, {month: 'long'}),
+      monthNameShort: date.toLocaleDateString(locale, {
+        month: 'short',
+      }),
+      day: date.getDate(),
+      dayName: date.toLocaleDateString(locale, {weekday: 'long'}),
+      dayNameShort: date.toLocaleDateString(locale, {
+        weekday: 'short',
+      }),
+      hour: date.getHours(),
+      minute: date.getMinutes(),
+      second: date.getSeconds(),
+      iso: date.toISOString(),
+    };
+  }
+
+  /**
+   * @param {!Date} date
+   * @param {string} locale
+   * @private
+   */
+  getVariablesInUTC_(date, locale) {
+    return {
+      year: date.getUTCFullYear(),
+      month: date.getUTCMonth() + 1,
+      monthName: date.toLocaleDateString(locale, {
+        month: 'long',
+        timeZone: 'UTC',
+      }),
+      monthNameShort: date.toLocaleDateString(locale, {
+        month: 'short',
+        timeZone: 'UTC',
+      }),
+      day: date.getUTCDate(),
+      dayName: date.toLocaleDateString(locale, {
+        weekday: 'long',
+        timeZone: 'UTC',
+      }),
+      dayNameShort: date.toLocaleDateString(locale, {
+        weekday: 'short',
+        timeZone: 'UTC',
+      }),
+      hour: date.getUTCHours(),
+      minute: date.getUTCMinutes(),
+      second: date.getUTCSeconds(),
+      iso: date.toISOString(),
+    };
+  }
+
+  /**
+   * @param {!Object} data
+   * @return {Object}
+   * @private
+   */
+  enhanceBasicVariables_(data) {
+    const hour12 = data.hour % 12 || 12;
+
+    return Object.assign({}, data, {
+      yearTwoDigit: this.padStart_(data.year % 100),
+      monthTwoDigit: this.padStart_(data.month),
+      dayTwoDigit: this.padStart_(data.day),
+      hourTwoDigit: this.padStart_(data.hour),
+      hour12,
+      hour12TwoDigit: this.padStart_(hour12),
+      minuteTwoDigit: this.padStart_(data.minute),
+      secondTwoDigit: this.padStart_(data.second),
+      dayPeriod: data.hour < 12 ? 'am' : 'pm',
+    });
+  }
+
+  /**
+   * @param {number} input
+   * @return {string}
+   * @private
+   */
+  padStart_(input) {
+    if (input > 9) {
+      return input.toString();
+    }
+
+    return '0' + input;
+  }
+
+  /**
+   * @param {!Element} element
+   * @private
+   */
+  rendered_(element) {
+    removeChildren(this.element);
```

Comment:
Yes, this was on purpose. The idea is that the transformation is a one-off. Should I change it?

Reply:
No this is good for now, but I wanted to make sure it was intentional. It would need to change if support for `amp-bind` is added later, but for this iteration it doesn't matter.